packages/tools/src/utilities/math/polyline/combinePolyline.ts

```diff
@@ -0,0 +1,350 @@
+import { Types } from '@cornerstonejs/core';
+import * as mathPoint from '../point';
+import getLineSegmentIntersectionsIndexes from './getLineSegmentIntersectionsIndexes';
+import containsPoint from './containsPoint';
+import getNormal2 from './getNormal2';
+import { glMatrix, vec3 } from 'gl-matrix';
+import getLineSegmentsIntersection from './getLineSegmentsIntersection';
+
+enum PolylinePointType {
+  Vertex,
+  Intersection,
+}
+
+// Position of the point related to the intersection region
+enum PolylinePointPosition {
+  Outside = -1,
+  Edge = 0,
+  Inside = 1,
+}
+
+// Direction from last point to the intersection point to know if it is entering
+// or exiting the intersection region
+enum PolylinePointDirection {
+  Exiting = -1,
+  Unknown = 0,
+  Entering = 1,
+}
+
+type PolylinePoint = {
+  type: PolylinePointType;
+  coordinates: Types.Point2;
+  position?: PolylinePointPosition;
+  visited: boolean;
+  next: PolylinePoint;
+};
+
+type PolylineIntersectionPoint = PolylinePoint & {
+  direction: PolylinePointDirection;
+  cloned?: boolean;
+};
+
+/**
+ * Ensure all polyline point objects are pointing to the next object in case
+ * it is still not point to anyone.
+ * @param polylinePoints - Array that contains all polyline points (vertices and intersections)
+ */
+function ensuresNextPointers(polylinePoints: PolylinePoint[]) {
+  // Make sure all nodes point to a valid node
+  for (let i = 0, len = polylinePoints.length; i < len; i++) {
+    const currentPoint = polylinePoints[i];
+
+    if (!currentPoint.next) {
+      currentPoint.next = polylinePoints[i === len - 1 ? 0 : i + 1];
+    }
+  }
+}
+
+/**
+ * Creates one linked list per polyline that contains all vertices and intersections
+ * found while walking along the edges.
+ *
+ * @param targetPolyline - Target polyline
+ * @param sourcePolyline - Source polyline
+ * @returns Two linked lists with all vertices and intersections.
+ */
+function getSourceAndTargetPointsList(
+  targetPolyline: Types.Point2[],
+  sourcePolyline: Types.Point2[]
+) {
+  const targetPolylinePoints: PolylinePoint[] = [];
+  const sourcePolylinePoints: PolylinePoint[] = [];
+  const sourceIntersectionsCache = new Map<
+    number,
+    PolylineIntersectionPoint[]
+  >();
+
+  const isFisrtPointInside = containsPoint(sourcePolyline, targetPolyline[0]);
+
+  let intersectionPointDirection = isFisrtPointInside
+    ? PolylinePointDirection.Exiting
+    : PolylinePointDirection.Entering;
```

Comment:
@sedghi This was the problem with that first point inside the polyline. It may also be on the edge but that requires a bigger change for a much lower probability of happening.